lib/virt_autotest/utils.pm

```diff
@@ -42,62 +42,72 @@ sub trim {
-        assert_script_run("kill -9 $_libvirtd_pid");
-        assert_script_run("$_libvirtd_cmd");
+#return 1 if test is expected to run on XEN hypervisor
+sub is_xen_host {
+    return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
+}
+
+# Usage: check_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be checked: check_modular_libvirt_daemons(qemu storage ...)
+# to check all required modular daemons without any daemons passed
+sub check_modular_libvirt_daemons {
+    my @daemons = @_;
+
+    if (!@daemons) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy lock);
+        # For details, please refer to poo#137096
+        (is_xen_host) ? push @daemons, 'xen' : push @daemons, ('qemu', 'log');
     }
-    else {
-        systemctl("restart libvirtd", timeout => 180);
+
+    foreach my $daemon (@daemons) {
+        systemctl("status virt${daemon}d.service");
+        if (($daemon == 'lock') || ($daemon == 'log')) {
+            systemctl("status virt${daemon}d\{,-admin\}.socket");
+        } else {
+            systemctl("status virt${daemon}d\{,-ro,-admin\}.socket");
+        }
     }
-    record_info("Libvirtd has been restarted!");
     save_screenshot;
-}
 
-#return 1 if test is expected to run on XEN hypervisor
-sub is_xen_host {
-    return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
+    record_info("Modular libvirt daemons checked, all active for", join(' ', @daemons));
 }
 
 # Usage: restart_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
 # to specify daemons which will be restarted: restart_modular_libvirt_daemons(virtqemud virtstoraged ...)
 # to restart all modular daemons without any daemons passed
 sub restart_modular_libvirt_daemons {
-    my @daemons;
+    my @daemons = @_;
 
-    if (@_ == 0) {
-        if (is_xen_host) {
-            @daemons = qw(virtxend virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
-        } else {
-            @daemons = qw(virtqemud virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
-        }
-    } else {
-        @daemons = @_;
+    if (!@daemons) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy lock);
+        # For details, please refer to poo#137096
+        (is_xen_host) ? push @daemons, 'xen' : push @daemons, ('qemu', 'log');
     }
 
     if (is_alp) {
         record_soft_failure("Restarting modular libvirt daemons has not been implemented in ALP. See poo#129086");
     } else {
         # Restart the sockets first
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon\{,-ro,-admin\}.socket");
+            return if (is_xen_host && is_tumbleweed && ($daemon == 'log'));
```

Comment:
@guoxuguang  Hi Leon, there is only one commit in the PR https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17551/commits. 
It is not easy to tell which lines are the new code. But I guess this section would have the core change. So I mainly check this part diff. 
- Based on https://bugzilla.suse.com/show_bug.cgi?id=1215809#c1, xen does not require virtlogd. So I think is_tumbleweed can be removed from this line.
- But I am not sure the purpose of `return if ...`. To me, `continue` is better.

Reply:
Good point,  refer to current log logic, virtlogd was not included both SLE and TW Xen host. 
```
  81     if (!@daemons) {
  82         @daemons = qw(network nodedev nwfilter secret storage proxy lock);
  83         # For details, please refer to poo#137096
  84         (is_xen_host) ? push @daemons, 'xen' : push @daemons, ('qemu', 'log');
  85     }
```
Keep this `return if (is_xen_host && is_tumbleweed && ($daemon == 'log')); `just only double check with virtlogd  on TW XEN host. 
For now, had removed it now. Thanks so much for your point.